### PR TITLE
feat: react-query example

### PR DESCRIPTION
**Pull request by n4ze3m**

### CODE REVIEW

File: with-react-query/search.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from "react"
+import { useQuery } from "react-query"
+
+type User = {
+  name: string
+  followers: number
+}
+
+const Search = () => {
+  const [username, setUsername] = React.useState(undefined)
```
Comment (louisgv) [inline with-react-query/search.tsx:10]: Mainly to align with other example, we should import `useState` specifically:

```
  const [username, setUsername] = useState<string>()
```

File: with-react-query/search.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const { status, error, data } = useQuery<boolean, Error, User>(
+    ["githubUser", username],
+    ({ queryKey }) =>
```
Comment (louisgv) [inline with-react-query/search.tsx:13]: Can we use an async callback instead? That'd make it a bit cleaner :D

**Review by louisgv — CHANGES_REQUESTED** [review]
Great PR 🔥🔥🔥- Just some minor comment, but otherwise LGTM

File: with-react-query/search.tsx (hunk 1 above)
Comment (n4ze3m) [reply with-react-query/search.tsx:10]: yes

File: with-react-query/search.tsx (hunk 2 above)
Comment (n4ze3m) [reply with-react-query/search.tsx]: yes

**Review by louisgv — APPROVED** [review]
LGTM 👍

**Pull request merged**